solr/core/src/java/org/apache/solr/handler/admin/ClusterStatus.java

```diff
@@ -92,11 +96,57 @@ public static Health combine(Collection<Health> states) {
   public ClusterStatus(ZkStateReader zkStateReader, ZkNodeProps props) {
     this.zkStateReader = zkStateReader;
     this.message = props;
-    collection = props.getStr(ZkStateReader.COLLECTION_PROP);
+    collectionParam = props.getStr(ZkStateReader.COLLECTION_PROP);
+    liveNodesParam = props.getBool(ZkStateReader.LIVENODES_PROP, false);
+    clusterPropertiesParam = props.getBool(ZkStateReader.CLUSTER_PROP, false);
+    rolesParam = props.getBool(ZkStateReader.ROLES_PROP, false);
+    includeAll = props.getBool(ZkStateReader.INCLUDE_ALL, true);
   }
 
   public void getClusterStatus(NamedList<Object> results)
       throws KeeperException, InterruptedException {
+
+    List<String> liveNodes = null;
+    NamedList<Object> clusterStatus = new SimpleOrderedMap<>();
+    if (includeAll || collectionParam != null || liveNodesParam) {
+      liveNodes =
+          zkStateReader.getZkClient().getChildren(ZkStateReader.LIVE_NODES_ZKNODE, null, true);
+      // add live_nodes
+      clusterStatus.add("live_nodes", liveNodes);
+    }
+
+    if (includeAll || collectionParam != null)
+      fetchClusterStatusForCollOrAlias(clusterStatus, liveNodes);
+
+    if (includeAll || clusterPropertiesParam) {
+      // read cluster properties
+      Map<String, Object> clusterProps = zkStateReader.getClusterProperties();
+      if (clusterProps != null && !clusterProps.isEmpty()) {
+        clusterStatus.add("properties", clusterProps);
+      }
+    }
+
+    // add the roles map
+    if (includeAll || rolesParam) {
```

Comment:
instead of constantly checking includeAll _as well as_ the one param for the category, consider defaulting the category boolean to be based on includeAll.  The way you coded it, it's impossible to include all and exclude one category.  I'm looking at the beginning of ColStatus.getColStatus for inspiration and seeing what I suggest would be really trivial; just change a bunch of "false" references to "includeAll"